src/generate.rs

```diff
@@ -138,10 +142,69 @@ pub fn generate_pair(opt: &GenerateOptions, rng: &mut impl Rng) -> (Sequence, Se
             let start = rng.gen_range(0..=b.len_bytes());
             b.insert(start, piece.as_str());
         }
+        ErrorModel::Repetition => {
+            let start = rng.gen_range(0..=b.len_bytes() - num_mutations);
+            let piece = b.slice(start..num_mutations).to_string();
+            let initial_length = b.len_bytes();
+            b = ropey::Rope::new();
+            // insert pieces
+            for _ in 0..initial_length / num_mutations {
+                b.insert(b.len_bytes(), piece.as_str());
+            }
+            // insert the segment of the last piece
+            b.insert(b.len_bytes(), &piece[initial_length - b.len_bytes()..]);
+        }
+        ErrorModel::Insert => {
+            let start = rng.gen_range(0..num_mutations - num_mutations);
+            let piece = b.slice(start..start + num_mutations).to_string();
+            b.insert(start, piece.as_str());
+        }
+        ErrorModel::Doubleinsert => {
+            let start = rng.gen_range(0..num_mutations);
+            let piece = b.slice(start..start + num_mutations * 2).to_string();
+            b.insert(start, piece.as_str());
+        }
+        ErrorModel::PiecesMutation => {
+            let mut b_initial = b.clone();
+            let mut b_substitutional = b.clone();
+            for i in 0..num_mutations {
+                for _ in 0..num_mutations {
+                    make_mutation(&mut b_substitutional, rng);
+                }
+                b.insert(b_initial.len_bytes() * i, &b_substitutional.to_string());
+                b_initial = b_substitutional.clone();
+            }
+        }
+        ErrorModel::RecursiveMutation => {
+            let mut b_substitutional = b.clone();
+            for i in 0..num_mutations {
+                for _ in 0..num_mutations {
+                    make_mutation(&mut b_substitutional, rng);
+                }
+                b.insert(
+                    b_substitutional.len_bytes() * i,
+                    &b_substitutional.to_string(),
+                );
+            }
+        }
```

Comment:
I don't think these two methods make sense as you wrote them now.

What I had in mind:

Start with some random piece X of some (possibly small) length.
Then make something like
```
A = YYYYYYY
B = YYYYYYYYY
```
where each `Y` is a mutation of `X`.


You can also accumulate mutations, so that you make a sequence of Y's and append them alternatingly to A and B or so.